uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/leave-request-ctrl_test.js

```diff
@@ -629,6 +629,16 @@
         });
 
         describe('save leave request', function () {
+          describe('allows multiple save', function() {
+            beforeEach(function() {
+              $ctrl.submit();
+            });
+
+            it('does not allow user to submit again', function () {
+              expect($ctrl.didSubmit).toBeTruthy();
```

Comment:
Delegating a data-integrity check mechanism to the view it's not a good idea. Use the flag on the view to show/hide the buttons, but put the actual multiple-saves-preventing mechanism in the controller